File: Fauzan Risqullah-SMKN 4 Tangerang/rest/rest/FrmPayment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Configuration;
using System.Data;
using System.Data.SqlClient;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace rest
{
    public partial class FrmPayment : Form
    {
        static string myconnstr = ConfigurationManager.ConnectionStrings["connstr"].ConnectionString;
        SqlConnection conn = new SqlConnection(myconnstr);

        public FrmPayment()
        {
            InitializeComponent();
        }

        private void FrmPayment_Load(object sender, EventArgs e)
        {
            string sql = "SELECT * FROM headerorder WHERE status = 'Memesan'";
            SqlCommand cmd = new SqlCommand(sql, conn);
            SqlDataAdapter sda = new SqlDataAdapter(cmd);

            DataTable dt = new DataTable();
            sda.Fill(dt);

            for (int i = 0; i < dt.Rows.Count; i++)
            {
                cmbID.Items.Add(dt.Rows[i]["orderid"].ToString());
            }
        }

        private void cmbID_SelectedIndexChanged(object sender, EventArgs e)
        {
            string sql = "SELECT name, qty, detailorders.price, total FROM detailorders INNER JOIN msmenu ON detailorders.menuid = msmenu.menuid WHERE orderid = '" + int.Parse(cmbID.Text) + "' AND status = 'Selesai'";

            SqlCommand cmd = new SqlCommand(sql, conn);
            SqlDataAdapter sda = new SqlDataAdapter(cmd);

            DataTable dt = new DataTable();
            sda.Fill(dt);

            dgv.DataSource = dt;
        }

        private void button4_Click(object sender, EventArgs e)
        {
            this.Hide();
        }

        private void btnSave_Click(object sender, EventArgs e)
        {
            string sql = "UPDATE headerorder SET date = '" + dtPicker.Value + "', payment = '" + cmbPayment.Text + "', bank = '" + cmbBank.Text + "', status = 'Paid'";
            SqlCommand cmd = new SqlCommand(sql, conn);

            conn.Open();

            int rows = cmd.ExecuteNonQuery();

            if (rows > 0)
            {
                MessageBox.Show("Success!", "Notice", MessageBoxButtons.OK, MessageBoxIcon.Information);
                this.Hide();
            }
            else
            {
                MessageBox.Show("Failed!", "Notice", MessageBoxButtons.OK, MessageBoxIcon.Warning);
            }

            conn.Close();
        }
    }
}

File: Daniel SMKS Bhaktianindya/SMKNUNSATAR/SMKNUNSATAR/FrmPayment.vb
Imports System.Data.SqlClient
Public Class FrmPayment
    Sub Tampildata()
        Call Koneksi()
        da = New SqlDataAdapter("Select Msmenu.Name as Menu, Detailorder.Qty From Detailorder inner join Msmenu on Msmenu.MenuID = Detailorder.MenuID Where OrderID = '" & ComboBox1.Text & "'", conn)
        ds = New DataSet
        da.Fill(ds, "Msmenu")
        DataGridView1.DataSource() = ds.Tables("Msmenu")
    End Sub

    Private Sub FrmPayment_Closed(sender As Object, e As EventArgs) Handles Me.Closed
        If CloseReason.ApplicationExitCall Then
            FrmCashierNav.Show()
        End If
    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        If (ComboBox2.Text = "") Or (TextBox2.Text = "") Or (ComboBox3.Text = "") Then
            MsgBox("Harap pilih data terlebih dahulu")
        Else
            Call koneksi()
            cmd = New SqlCommand("Update Headerorder set Payment = '" & ComboBox2.Text & "', Bank = '" & ComboBox3.Text & "'", conn)
            cmd.ExecuteNonQuery()
            MsgBox("Data berhasil disave")
        End If
    End Sub

    Private Sub FrmPayment_Load(sender As Object, e As EventArgs) Handles Me.Load
        Call Koneksi()
        cmd = New SqlCommand("Select * From Headerorder", conn)
        dr = cmd.ExecuteReader
        While dr.Read()
            Dim orderid = dr.Item(0)
            ComboBox1.Items.Add(orderid)
        End While
        Call Tampildata()
    End Sub
End Class